### ISSUE TITLE
How to use patterns in Validator

### ISSUE THREAD

**Issue by aflopes** [rochacbruno/dynaconf#587]
The class `Validator` documents that conditions can be attached to variable names or patterns

```python
class Validator:
    """Validators are conditions attached to settings variables names
    or patterns::

    ...
        `names` are a one (or more) names or patterns::
        Validator('NAME')
        Validator('NAME', 'OTHER_NAME', 'EVEN_OTHER')
        Validator(r'^NAME', r'OTHER./*')

    ...
    """
```

How can we use patterns to apply a condition to several defined settings? Is there any example available?

**Comment by ajossorioarana** [comment]
Hi! From the modifications in the milestonews I get that this is a feature that has not been implemented yet. 

Is that so? (I'm not well versed on GitHub repositories, forgive my ignorance!) If it is currently a feature, I also ask for some help on how to use patterns to apply a condition.

**Comment by rochacbruno** [comment]
@ajossorioarana you are right, for some reason patterns have been mentioned on the docstring but not implemented on the code 
https://github.com/dynaconf/dynaconf/blob/master/dynaconf/validator.py#L228 

The `self.names` variable will need to be first resolved using the patterns and the regex module.

Right now the only workaround I can think is to add a `dynaconf_hooks.py` file and then validate there.

That is why I moved this to 3.2.0, needs to be fixed in the next minor version as a bugfix, not a feature.

**Comment by rochacbruno** [comment]
Now with schema on #1107 we are moving to a different way to use validators based on type annotations.

I think, we must remove any mentioning of patterns from docstring/docs because this feature is not supported.

And add to the docs/tests_functional one example using a dynaconf post hook to solve the problem.
